### PR TITLE
chore: add flake8 ci and precommit checks

### LINKED ISSUE #42: Add pylint to precommit and CI
**Is your feature request related to a problem? Please describe.**

Black is just a formatter and does not catch things like unused imports.

**Describe the solution you'd like**

pylint/flake8 to precommit hook and CI.

### PR DESCRIPTION
**Pull request by iamogbz**
Resolves #42 

Closes #47

### CODE REVIEW

File: src/syrupy/serializers/yaml.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ def file_extension(self) -> str:
     def discover_snapshots(self, filepath: str) -> Set[str]:
         try:
             return set(self._read_file(filepath).keys())
-        except:
+        except:  # noqa: E722
```
Comment (noahnu) [inline src/syrupy/serializers/yaml.py]: Does flake offer more descriptive exceptions? pylint offers code & human readable string

Comment (iamogbz) [reply src/syrupy/serializers/yaml.py]: Yeah couldn't find any which was annoying

File: .flake8 [hunk 2]
```diff
@@ -0,0 +1,2 @@
+[flake8]
+max-line-length=88
```
Comment (noahnu) [inline .flake8]: Is there an open issue for project toml support?

File: src/syrupy/session.py [hunk 3]
```diff
@@ -99,13 +94,15 @@ def finish(self) -> None:
                     if not count:
                         continue
                     path_to_file = os.path.relpath(filepath, self.base_dir)
+                    deleted_snapshots = ", ".join(map(bold, sorted(snapshots)))
```
Comment (noahnu) [inline src/syrupy/session.py]: Interesting. I think pylint prefers comprehension over map

File: src/syrupy/session.py [hunk 4]
```diff
@@ -99,13 +94,15 @@ def finish(self) -> None:
...
                     self.add_report_line(
-                        f"Deleted {', '.join(map(bold, sorted(snapshots)))} ({path_to_file})"
+                        f"Deleted {deleted_snapshots} ({path_to_file})"
                     )
             else:
                 self.add_report_line(
                     gettext(
-                        "Re-run pytest with --snapshot-update to delete the unused snapshots."
+                        "Re-run pytest with --snapshot-update"
+                        "to delete the unused snapshots."
```
Comment (noahnu) [inline src/syrupy/session.py]: Missing a space

File: .flake8 (hunk 2 above)
Comment (iamogbz) [reply .flake8]: https://gitlab.com/pycqa/flake8/issues/428

File: src/syrupy/session.py (hunk 4 above)
Comment (iamogbz) [reply src/syrupy/session.py]:
```suggestion
                        " to delete the unused snapshots."
```

**Pull request merged**